tests/helper/helper_generic.go

```diff
@@ -312,12 +317,48 @@ func CommonBeforeEach() CommonVar {
 
 // CommonAfterEach is common function that cleans up after every test Spec (It)
 func CommonAfterEach(commonVar CommonVar) {
+	// Get details, including test result for each test spec and adds it to local testResults.txt file
+	// Ginkgo test related variables.
+	commonVar.testFileName = strings.Replace(CurrentGinkgoTestDescription().FileName[strings.LastIndex(CurrentGinkgoTestDescription().FileName, "/")+1:strings.LastIndex(CurrentGinkgoTestDescription().FileName, ".")], "_", "-", -1) + ".go"
+	commonVar.testCase = CurrentGinkgoTestDescription().FullTestText
+	commonVar.testFailed = CurrentGinkgoTestDescription().Failed
+	commonVar.testDuration = CurrentGinkgoTestDescription().Duration.Seconds()
+
+	var prNum string
+	var resultsRow string
+	prNum = os.Getenv("GIT_PR_NUMBER")
+	passedOrFailed := "PASSED"
+	if commonVar.testFailed {
+		passedOrFailed = "FAILED"
+	}
+	clusterType := "OCP"
+	if IsKubernetesCluster() {
+		clusterType = "KUBERNETES"
+	}
+	testDate := strings.Split(time.Now().Format(time.RFC3339), "T")[0]
+	resultsRow = prNum + "," + testDate + "," + clusterType + "," + commonVar.testFileName + "," + commonVar.testCase + "," + passedOrFailed + "," + strconv.FormatFloat(commonVar.testDuration, 'E', -1, 64) + "\n"
+	testResultsFile := filepath.Join("/", "tmp", "testResults.txt")
+
+	f, err := os.OpenFile(testResultsFile, os.O_APPEND|os.O_WRONLY|os.O_CREATE, 0600)
+	if err != nil {
+		fmt.Println("Error: ", err)
+	} else {
+		f.Close()
+		_, err = f.WriteString(resultsRow)
+		if err != nil {
+			fmt.Println("Error: ", err)
+		}
+		if err = f.Close(); err != nil {
+			fmt.Println("Error: ", err)
+		}
```

Comment:
I think if you do what Dharmit suggested in this [comment,](https://github.com/redhat-developer/odo/pull/5449/files#r818389371) I think you can remove this portion.

Reply:
followed Parthvi's suggestion as the defer was causing unit tests to fail. Now last thing I do in the after each is closing the file.